src/coding-guidelines/expressions.rst

```diff
@@ -7,6 +7,269 @@ Expressions
+        :id: rat_LvrS1jTCXEOk
+        :status: draft
+
+        Eliminate arithmetic overflow to avoid runtime panics and unexpected wraparound behavior.
+        Use explicit wrapping or saturation semantics in cases where these behaviors are intentional.
+        Range checking can be used to eliminate the possibility of arithmetic overflow.
+
+    .. non_compliant_example::
+        :id: non_compl_ex_cCh2RQUXeH0N
+        :status: draft
+
+        This noncompliant code example can result in arithmetic overflow during the addition of the signed operands ``si_a`` and ``si_b``:
+
+        .. code-block:: rust
+
+            fn add(si_a: i32, si_b: i32) {
+              let sum: i32 = si_a + si_b;
+              // ...
+            }
+
+    .. compliant_example::
+        :id: compl_ex_BgUHiRB4kc4b_1
+        :status: draft
+
+        This compliant solution ensures that the addition operation cannot result
+        in arithmetic overflow, based on the maximum range of a signed 32-bit integer.
+        A more restrictive range may also be used.
+
+        .. code-block:: rust
+
+            enum ArithmeticError {
+                Overflow,
+	            DivisionByZero,
+            }
+
+            use std::i32::{MAX as INT_MAX, MIN as INT_MIN};
+
+	        fn add(si_a: i32, si_b: i32) -> Result<i32, ArithmeticError> {
+                if (si_b > 0 && si_a > INT_MAX - si_b)
+                    || (si_b < 0 && si_a < INT_MIN - si_b)
+                {
+                    Err(ArithmeticError::Overflow)
+                } else {
+                    Ok(si_a + si_b)
+                }
+            }
+
+            fn sub(si_a: i32, si_b: i32) -> Result<i32, ArithmeticError> {
+                if (si_b < 0 && si_a > INT_MAX + si_b)
+                    || (si_b > 0 && si_a < INT_MIN + si_b)
+                {
+                   Err(ArithmeticError::Overflow)
+                } else {
+                   Ok(si_a - si_b)
+                }
+            }
+
+	        fn mul(si_a: i32, si_b: i32) -> Result<i32, ArithmeticError> {
+                if si_a == 0 || si_b == 0 {
+                    return Ok(0);
+                }
+
+                // Detect overflow before performing multiplication
+                if (si_a == -1 && si_b == INT_MIN) || (si_b == -1 && si_a == INT_MIN) {
+                    Err(ArithmeticError::Overflow)
+                } else if (si_a > 0 && (si_b > INT_MAX / si_a || si_b < INT_MIN / si_a))
+                    || (si_a < 0 && (si_b > INT_MIN / si_a || si_b < INT_MAX / si_a))
+                {
+                    Err(ArithmeticError::Overflow)
+                } else {
+                    Ok(si_a * si_b)
+                }
+            }
+
+    .. compliant_example::
+        :id: compl_ex_BgUHiRB4kc4c
+        :status: draft
+
+        This compliant example uses safe checked addition instead of manual bounds checks.
+        Checked functions can reduce readibility when complex arithmetic expressions are needed.
```

Comment:
```suggestion
        Checked functions can reduce readability when complex arithmetic expressions are needed.
```

I'm suggesting this change real quick so that the build passes. Full review coming in a bit